tests/serial/time/implicit/parallel-coupling/WaveformSubcyclingWithDifferentNumberOfDts.cpp

```diff
@@ -0,0 +1,119 @@
+#ifndef PRECICE_NO_MPI
+
+#include "testing/Testing.hpp"
+
+#include <precice/precice.hpp>
+#include <vector>
+
+using namespace precice;
+
+BOOST_AUTO_TEST_SUITE(Integration)
+BOOST_AUTO_TEST_SUITE(Serial)
+BOOST_AUTO_TEST_SUITE(Time)
+BOOST_AUTO_TEST_SUITE(Implicit)
+BOOST_AUTO_TEST_SUITE(ParallelCoupling)
+
+/**
+ * @brief Test to run a simple coupling where the number of time steps changes with the waveform iteration.
+ *
+ * Provides a dt argument to the read function. A first order waveform is used.
+ */
+BOOST_AUTO_TEST_CASE(WaveformSubcyclingWithDifferentNumberOfDts)
+{
+  PRECICE_TEST("SolverOne"_on(1_rank), "SolverTwo"_on(1_rank));
+
+  Participant precice(context.name, context.config(), 0, 1);
+
+  typedef double (*DataFunction)(double);
+
+  // The functions that are exchanged in iteration 1 and 2 respectively.
+  DataFunction funcIterOne = [](double t) -> double {
+    return (double) (1 + 2 * t) * (2 + 3 * t) * (3 + 4 * t);
+  };
+  DataFunction funcIterTwo = [](double t) -> double {
+    return (double) (6 + 5 * t) * (1 + 6 * t) * (1 + 6 * t);
+  };
+
+  BOOST_TEST(funcIterOne(0) == funcIterTwo(0)); // because the data at WINDOW_START does not change over iterations.
+
+  int         nTimeSteps;
+  std::string meshName, writeDataName, readDataName;
+  if (context.isNamed("SolverOne")) {
+    meshName      = "MeshOne";
+    writeDataName = "DataOne";
+    readDataName  = "DataTwo";
+    nTimeSteps    = 6;
+  } else {
+    BOOST_TEST(context.isNamed("SolverTwo"));
+    meshName      = "MeshTwo";
+    writeDataName = "DataTwo";
+    readDataName  = "DataOne";
+    nTimeSteps    = 4;
+  }
+
+  double   writeData, readData;
+  VertexID vertexID;
+
+  double v0[] = {0, 0, 0};
+  vertexID    = precice.setMeshVertex(meshName, v0);
+  int    timestepCheckpoint;
+  double time = 0;
+
+  if (precice.requiresInitialData()) {
+    writeData = funcIterOne(time);
+    precice.writeData(meshName, writeDataName, {&vertexID, 1}, {&writeData, 1});
+  }
+
+  precice.initialize();
+  double TimeWindowSize = precice.getMaxTimeStepSize();
```

Comment:
```suggestion
  double windowDt = precice.getMaxTimeStepSize();
```
Please try to keep things consistent with other tests. 

https://github.com/precice/precice/blob/f1adf55c6a34bb7b6e87e7691f69785cf0558741/tests/serial/time/implicit/serial-coupling/ReadWriteScalarDataWithWaveformSamplingFirst.cpp#L74-L79.

Reply:
Btw: I'm surprised that the linter does not detect this. @NiklasKotarsky Could you please open an issue?

Reply:
Yes, do you mean the fact that I was able to write TimeWindowSize instead of timeWindowSize?

Reply:
Exactly. Variable names (in our c++ part of the code base) should be `timeWindowSize` (camelCase, starting with lower case).